histolab/tiler.py

```diff
@@ -1121,3 +1121,288 @@ def _scores(
             scores.append((score, tile_wsi_coords))
 
         return scores
+
+
+class ThresholdTiler(GridTiler):
```

Comment:
Sorry, but I d not understand why this is a subcalss of the `GridTiler`. `extract` and `_tiles_generator` have been reimplemented and no other methods or properties from the `GridTiler` have been used. Furthermore you have also changed the init method, so personally I find weird the `GridTiler` dependency. At a first glance it seems more similar to the `ScoreTiler`.

Reply:
> Sorry, but I d not understand why this is a subcalss of the `GridTiler`. `extract` and `_tiles_generator` have been reimplemented and no other methods or properties from the `GridTiler` have been used.

The reason is a subclass of GridTiler is because ThresholdTiler first extracts tiles like a GridTiler (calling its method `_tiles_generator` line 1402 of `tiler.py`), then for each tile it assigns a related score (line 1406), and finally all tiles that have a score less than a specified threshold are discarded.

> At a first glance it seems more similar to the `ScoreTiler`.

Did you mean inheriting from `ScoreTiler`?
If yes, the problem with inheriting from `ScoreTiler` is that, when instantiating a `ThresholdTiler`, the user must supply `n_tiles` as a parameter, which I believe is questionable for a `ThresholdTiler` since the user does not know in advance how many tiles must be removed (he's more interested in all tiles that have a score higher than a threshold).

Reply:
gotcha